File: DSA/DivyanshKumarSingh_590012175/Day_41_Question_1.c
#include <stdio.h>
#include <string.h>

#define MAX 1000

void removeAdjacentDuplicates(char *str)
{
    char stack[MAX];
    int top = -1;

    for (int i = 0; str[i]; i++)
    {
        if (top >= 0 && stack[top] == str[i])
        {
            top--; // Remove duplicate
        }
        else
        {
            stack[++top] = str[i];
        }
    }

    stack[top + 1] = '\0';
    printf("Reduced String: %s\n", stack);
}

int main()
{
    char str1[] = "abbaca";
    char str2[] = "azxxzy";

    printf("Original: %s\n", str1);
    removeAdjacentDuplicates(str1);

    printf("Original: %s\n", str2);
    removeAdjacentDuplicates(str2);

    return 0;
}

File: DSA/Pranvkumar_590011587/Day41_Question1.py
def remove_adjacent_duplicates(s: str) -> str:
    while True:
        stack = []
        i = 0
        while i < len(s):
            if stack and stack[-1] == s[i]:
                stack.pop()  
            else:
                stack.append(s[i])
            i += 1
        new_s = ''.join(stack)
        if new_s == s:  
            break
        s = new_s
    return s

# Example Usage
print(remove_adjacent_duplicates("abbaca")) 
print(remove_adjacent_duplicates("azxxzy"))  